allennlp/service/servable/models/semantic_role_labeler.py

```diff
@@ -55,3 +39,23 @@ def predict_json(self, inputs: JSONDict) -> JSONDict:
                 })
 
         return results
+
+    @classmethod
+    def from_params(cls, params: Params) -> 'SemanticRoleLabelerServable':
+        tokenizer = Tokenizer.from_params(params.pop("tokenizer"))
+
+        token_indexers = {}
+        token_indexer_params = params.pop('token_indexers')
+        for name, indexer_params in token_indexer_params.items():
+            token_indexers[name] = TokenIndexer.from_params(indexer_params)
+
+        vocab_dir = params.pop('vocab_dir')
+        vocab = Vocabulary.from_files(vocab_dir)
```

Comment:
Shouldn't this just be the model serialization prefix?  Presumably you're going to want to load a model here, too.

I'm wondering if there's just an easier way to do this.  If you load the experiment configuration file, you can pull out the model params, instantiate the model (which already has a vocab dir), pull out the dataset reader and grab its tokenizer and token indexers, and not have to do all of the individual processing and instantiation that's done here.  That way you don't have to duplicate the configuration files or the loading code, you just pull out the pieces you need after the already-existing code runs.

And if we go that route, I don't think I'd call this method `from_params`, I'd probably call it `from_experiment_config`, or `from_experiment_params`, or something.  There's no need to have a `from_params` method for this if it doesn't make sense.  The models and things themselves should definitely be instantiated `from_params`, but that doesn't mean this class has to match the same API.

What do you think?